pkg/trace/writer/trace.go

```diff
@@ -139,6 +145,49 @@ func (w *TraceWriter) Run() {
 	}
 }
 
+func (w *TraceWriter) runSync() {
+	defer close(w.stop)
+	defer close(w.flushChan)
+	for {
+		select {
+		case pkg := <-w.In:
+			w.addSpans(pkg)
+		case notify := <-w.flushChan:
```

Comment:
Seems like the only difference between the two methods (`runSync` and `runAsync`) is the `flushChan` case, so we can maybe stick to just one method `Run`. Just like we had before, and just like you did in the stats writer? There would be an extra case for `flushChan` but that wouldn't be triggered if sync mode is off. WDYT? Or did I miss something?

Reply:
The runAsync also uses a ticker for flushing, which we don't want in the runSync loop. That's the only other difference, but it meant we couldn't use the same select statement for both.

Reply:
Ah yes. Doh. Sorry I missed that. Resolving.